src/wasm-binary.h

```diff
@@ -836,6 +837,7 @@ class WasmBinaryBuilder {
   std::map<Index, std::vector<CallImport*>> functionImportCalls; // at index i we have all callImports to the imported function i
   Function* currFunction = nullptr;
   Index endOfFunction = -1; // before we see a function (like global init expressions), there is no end of function to check
+  inline bool isInFunction() { return endOfFunction != (Index)-1; }
```

Comment:
i think it would be better to check `currFunction`. but also can assert on `endOfFunction`.

Reply:
`currFunction` is after most of its header is parsed (after locals), I need to determine if function parsing started before makeFunction is called.

Reply:
Not sure I understand (how can function parsing start before the function?), but can we set currFunction earlier? It seems like that should match whether we are in a function or not, at least I think that was the intention when it was added originally. If not, perhaps we should rename it?